config

```diff
@@ -1,937 +1,854 @@
+        # HPUX 10 and 11 targets are unified
+        return "${MACHINE}-hp-hpux1x" if $HPUXVER =~ m@1[0-9]@;
+        return "${MACHINE}-hp-hpux";
+    } ],
+
+    [ 'BSD/386:.*:.*:.*486.*|BSD/OS:.*:.*:.*:.*486', sub {
+        my $BSDVAR = `/sbin/sysctl -n hw.model`;
+        return "i586-whatever-bsdi" if $BSDVAR =~ m@Pentium@;
+        return "i386-whatever-bsdi";
+    } ],
+    [ 'FreeBSD:.*:.*:.*386', sub {
+        my $VERS = $RELEASE;
+        $VERS = s/[-(].*//;
+        my $MACH = `sysctl -n hw.model`;
+        $MACH = "i386" if $MACH =~ m@386@;
+        $MACH = "i486" if $MACH =~ m@486@;
+        $MACH = "i686" if $MACH =~ m@Pentium II@;
+        $MACH = "i586" if $MACH =~ m@Pentium@;
+        $MACH = "$MACHINE" if $MACH !~ /i.86/;
+        my $ARCH = 'whatever';
+        $ARCH = "pc" if $MACH =~ m@i[0-9]86@;
+        return "${MACH}-${ARCH}-freebsd${VERS}";
+    } ],
+
+    [ 'NetBSD:.*:.*:.*386', sub {
+        #return "`(/usr/sbin/sysctl -n hw.model || /sbin/sysctl -n hw.model) | sed 's,.*\(.\)86-class.*,i\186,'`-whatever-netbsd";
+    } ],
+
+    [ 'OSF1:.*:.*:.*alpha', sub {
+        my $OSFMAJOR = $RELEASE;
+        $OSFMAJOR =~ 's/^V([0-9]*)\..*$/\1/';
+        return "${MACHINE}-dec-tru64" if $OSFMAJOR =~ m@[45]@;
+        return "${MACHINE}-dec-osf";
+    } ],
+
+    [ 'Darwin:', sub {
+        return "ppc-apple-darwin" if $MACHINE =~ m@Power@;
+        return "x86_64-apple-darwin" if $MACHINE =~ m@x86_64@;
+        return "i686-apple-darwin";
+    } ],
+];
+
+# Return the cputype-vendor-osversion
+sub get_sys_type {
+
+    # Special-cases for ISC, SCO, Unixware
+    my $REL = &is_sco_uname();
+    if ( $REL ne "" ) {
+        my $result = &get_sco_type($REL);
+        if ( $result ne '' ) {
+            $result =~ s/\$\{MACHINE\}/${MACHINE}/;
+            return $result;
+        }
+    }
+
+    # Now pattern-match
+    my $sys = "${SYSTEM}:${RELEASE}:${VERSION}:${MACHINE}";
+
+    # Simple cases
+    foreach my $tuple ( @$simple_sys_list ) {
+        my $pat = @$tuple[0];
+        next if $sys !~ /$pat/;
+        my $result = @$tuple[1];
+        $result =~ s/\$\{MACHINE\}/${MACHINE}/;
+        return $result;
+    }
+
+    # Complex cases.
+    foreach my $tuple ( @$complex_sys_list ) {
+        my $pat = @$tuple[0];
+        next if $sys !~ /$pat/;
+        my $ref = @$tuple[1];
+        my $result = &$ref;
+        $result =~ s/\$\{MACHINE\}/${MACHINE}/;
+        return $result;
+    }
+
+    # Oh well.
+    return "${MACHINE}-whatever-${SYSTEM}";
+}
+
+my $GUESSOS = get_sys_type();
+print "Operating system: $GUESSOS\n";
+$options .= " 386" if $GUESSOS =~ /i386-/;
+
+##
+##  SKIP ALGOLRITHMS THAT DON'T EXIST. XXX NEEDED?
+my @cryptodir = (
+    'aes', 'aria', 'bf', 'camellia', 'cast', 'des', 'dh', 'dsa', 'ec', 'hmac',
+    'idea', 'md2', 'md5', 'mdc2', 'rc2', 'rc4', 'rc5', 'ripemd', 'rsa',
+    'seed', 'sha', 'sm2', 'sm3', 'sm4'
+);
+foreach my $d ( @cryptodir ) {
+    $options .= " no-$d" if ! -d "$THERE/crypto/$d";
+}
```

Comment:
Oh wow... This is really old stuff, hardly relevant any more. Or does anyone still do that, remove whole crypto subsystems from the source?

Reply:
I very much suspect the answer is _yes_.

Reply:
That check should be entirely in Configure, then...

Reply:
I'll leave it now, and add a comment saying remove when this is moved into configure.